File: examples/config_test.py
import asyncio
from secureshell import SecureShell

async def main():
    # Initialize shell (will load secureshell.yaml)
    # Use None provider so gatekeeper is disabled (testing config bypass)
    shell = SecureShell(provider=None)
    shell.config.debug_mode = True
    
    print("🚀 Testing Whitelist/Blacklist Logic\n")
    print(f"Loaded config - Allowlist: {shell.config.allowlist}")
    print(f"Loaded config - Blocklist: {shell.config.blocklist}\n")
    
    # 1. Test Whitelist (Should bypass everything - no gatekeeper needed)
    print("--- Test 1: Whitelisted Command ---")
    res1 = await shell.execute("rm safe_delete.txt", "cleaning up")
    print(f"Success: {res1.success}")
    print(f"Risk Tier: {res1.risk_tier}")
    print(f"Reason: {res1.denial_reason or 'Command executed (or failed naturally)'}\n")

    # 2. Test Blacklist (Should be blocked immediately)
    print("--- Test 2: Blacklisted Command ---")
    res2 = await shell.execute("echo blocked_message", "just echoing")
    print(f"Success: {res2.success}")
    print(f"Reason: {res2.denial_reason}\n")
    
    # 3. Test Normal (Will be blocked due to no gatekeeper)
    print("--- Test 3: Normal Command (No Gatekeeper) ---")
    res3 = await shell.execute("echo normal_message", "just echoing")
    print(f"Success: {res3.success}")
    print(f"Reason: {res3.denial_reason or 'N/A'}")
    
    await shell.shutdown()

if __name__ == "__main__":
    asyncio.run(main())

File: examples/secureshell-ts/config_test.ts
/**
 * Config Test - YAML Configuration Demo for SecureShell TypeScript
 * 
 * Demonstrates allowlist/blocklist feature with command type matching.
 * 
 * Usage:
 *   npx tsx examples/secureshell-ts/features/config_test.ts
 */

import { SecureShell } from '@secureshell/ts';

async function main() {
    console.log('🚀 Testing Allowlist/Blocklist with Command Types\n');

    // Initialize shell without provider (testing config only)
    const shell = new SecureShell({
        config: {
            debugMode: true,
            allowlist: ['echo', 'ls', 'dir'],
            blocklist: ['rm', 'dd']
        }
    });

    console.log('Loaded config:');
    console.log(`Allowlist: ${JSON.stringify(shell.config.allowlist)}`);
    console.log(`Blocklist: ${JSON.stringify(shell.config.blocklist)}\n`);

    // Test 1: Allowlisted command type
    console.log('--- Test 1: Allowlisted Command Type ---');
    const res1 = await shell.execute('echo "This should work"', 'testing allowlist');
    console.log('Success:', res1.success);
    console.log('Risk Tier:', res1.risk_tier);
    console.log('Output:', res1.success ? res1.stdout : res1.stderr, '\n');

    // Test 2: Blocklisted command type
    console.log('--- Test 2: Blocklisted Command Type ---');
    const res2 = await shell.execute('rm some_file.txt', 'trying to delete');
    console.log('Success:', res2.success);
    console.log('Reason:', res2.stderr, '\n');

    // Test 3: Another allowlisted command
    console.log('--- Test 3: Another Allowlisted Command ---');
    const res3 = await shell.execute('dir', 'listing files');
    console.log('Success:', res3.success);
    console.log('Output:', res3.success ? res3.stdout.substring(0, 100) : res3.stderr);

    await shell.close();
}

main().catch(console.error);
